### ISSUE TITLE
macOS: Building Development and PhysX5Support branch builds fails due to failed AzFramework.Tests and AzToolsFramework.Tests  commands

### ISSUE THREAD

**Issue by LB-MateuszSzweda** [o3de/o3de#13697]
**Describe the bug**
CMake fails to build Development and PhysX5Support branches on MacBook Pro 2020 and MacBook Air 2020 macOS machines due to the following errors:

**MacBook Air 2020**
```
** BUILD FAILED **

The following build commands failed:
	CompileC /Users/macpro15/o3de/osx_build/build/AzFramework.Tests.build/profile/Objects-normal/x86_64/unity_1_cxx.o /Users/macpro15/o3de/osx_build/Code/Framework/AzFramework/CMakeFiles/AzFramework.Tests.dir/Unity/unity_1_cxx.cxx normal x86_64 c++ com.apple.compilers.llvm.clang.1_0.compiler (in target 'AzFramework.Tests' from project 'O3DE')
(1 failure)
Elapsed time: 882 s. (time), 0.060312 s. (clock)
```
**MacBook Pro 2020**
```
** BUILD FAILED **

The following build commands failed:
	CompileC /Users/macpro13/o3de/osx_build/Code/Framework/AzToolsFramework/O3DE.build/profile/AzToolsFramework.Tests.build/Objects-normal/x86_64/unity_10_cxx.o /Users/macpro13/o3de/osx_build/Code/Framework/AzToolsFramework/CMakeFiles/AzToolsFramework.Tests.dir/Unity/unity_10_cxx.cxx normal x86_64 c++ com.apple.compilers.llvm.clang.1_0.compiler (in target 'AzToolsFramework.Tests' from project 'O3DE')
(1 failure)
Elapsed time: 8845 s. (time), 0.129916 s. (clock)
```

The exact software specifications of the machines are:

* MacBook Pro 2020 (Intel Core i5, Big Sur 11.3.1,XCode 13.0, Xcode 13.2.1, CMake 3.22.0 on AutomatedTesting and New projects
* MacBook Air 2020 (M1 chipset, Ventura 13.0.1 , XCode 14.2 , CMake 3.25.0 on AutomatedTesting and New projects

Please note that prior to updating MacBook Air 2020 from  (M1 chipset, Monterey 12.6 , XCode 14.0.1 , CMake 3.24.3) to
 (M1 chipset, Ventura 13.0.1 , XCode 14.2 , CMake 3.25.0) on Development (35ee631) and PhysX5Support (c885bb7) this particular machine was able to build the Editor but it was crashing on launch with _RPI System Could not initialize_ correctly error,
however MacBook Pro 2020 machine was already failing the building with AzToolsFramework.Tests then.

For further information please refer to the attached failed build logs and failed launch logs.

**Steps to reproduce**
1. Obtain the Development branch build via GitHub.
2. Create a new or move your preexisting 3rdParty folder containing CMake into the o3de folder.
3. Obtain python packages with the _get_python.sh_ script located in the _o3de/python_ directory.
4. Create the _osx_build_ folder under the o3de location.
5. Set the following temporary environment variables in the Terminal:
```
export PATH=$PATH:/Users/<home_folder_name>/o3de/3rdParty/CMake/CMake.app/Contents/bin
export LY_3RDPARTY_PATH=/Users/<home_folder_name>/o3de/3rdParty
export LY_PACKAGE_SERVER_URLS=https://d2c171ws20a1rv.cloudfront.net
```
6. Configure and generate the Xcode solution with the following command:
`cmake .. -G "Xcode" -DLY_3RDPARTY_PATH=$LY_3RDPARTY_PATH -DLY_PROJECTS=AutomatedTesting -DLY_UNITY_BUILD=1`

7. Build Mac binaries with the following command:
`cmake -E time cmake --build . --target ALL_BUILD --config profile`

**Expected behavior**
Building process finishes successfully without errors.

**Actual behavior**
Building fails.

**Found in Branch**
Development

**Commit ID from [o3de/o3de](https://github.com/o3de/o3de) Repository**
976601b

**Desktop/Device (please complete the following information):**
- Device: Mac
- OS: macOS
- Version: Ventura 13.0.1
- CPU: M1
- GPU: M1
- Memory: 16GB

**Additional context**
[Development  976601b failed build log MacBook Air.txt](https://github.com/o3de/o3de/files/10228197/Development.976601b.failed.build.log.MacBook.Air.txt)
[Development_976601b_building_failed_log MacBookPro.txt](https://github.c
[PhysX5Support 83d53ad branch failed build log MacBook Air.txt](https://github.com/o3de/o3de/files/10228294/PhysX5Support.83d53ad.branch.failed.build.log.MacBook.Air.txt)
om/o3de/o3de/files/102
[Editor PhysXSupport5(c885bb7) building but not launching prior to update terminal log MacBook Air.txt](https://github.com/o3de/o3de/files/10228297/Editor.PhysXSupport5.c885bb7.building.but.not.launching.prior.to.update.terminal.log.MacBook.Air.txt)
28199/Development_976601b_building_failed_log.MacBookPro.txt)

**Comment by moraaar** [comment]
Hi, I've looked into this, there are several points to mention:

1. The "Development 976601b failed build log MacBook Air.txt" log shows **that specific machine has permission errors** when running CMake configuration and it's not configuring correctly. Maybe the user doesn't have enough permissions or the way CMake is installed and used doesn't have enough permissions. That's something to fix in that specific machine.
````
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed
CMake Error at /Applications/CMake.app/Contents/share/cmake-3.25/Modules/Internal/CheckSourceCompiles.cmake:127 (file):
  file failed to open for writing (Permission denied):

    /Users/macpro15/o3de/osx_build/CMakeFiles/CMakeError.log
Call Stack (most recent call first):
  /Applications/CMake.app/Contents/share/cmake-3.25/Modules/CheckCSourceCompiles.cmake:76 (cmake_check_source_compiles)
  /Applications/CMake.app/Contents/share/cmake-3.25/Modules/FindThreads.cmake:97 (CHECK_C_SOURCE_COMPILES)
  /Applications/CMake.app/Contents/share/cmake-3.25/Modules/FindThreads.cmake:165 (_threads_check_libc)
  /Applications/CMake.app/Contents/share/cmake-3.25/Modules/CMakeFindDependencyMacro.cmake:47 (find_package)
  3rdParty/packages/OpenEXR-3.1.3-rev4-mac/OpenEXR/lib/cmake/OpenEXR/OpenEXRConfig.cmake:32 (find_dependency)
  3rdParty/packages/OpenEXR-3.1.3-rev4-mac/o3de_package_utilities.cmake:136 (find_package)
  3rdParty/packages/OpenEXR-3.1.3-rev4-mac/FindOpenEXR.cmake:29 (o3de_import_existing_config_files)
  cmake/LYWrappers.cmake:587 (find_package)
  cmake/LYWrappers.cmake:266 (ly_parse_third_party_dependencies)
  Gems/Atom/Asset/ImageProcessingAtom/Code/CMakeLists.txt:39 (ly_add_target)
````

2. The "PhysX5Support.83d53ad.branch.failed.build.log.MacBook.Air.txt" log shows the following compilation error in a test file:
````
/Users/macpro15/o3de/Code/Framework/AzFramework/Tests/FileIO.cpp:179:23: error: implicit conversion loses integer precision: 'clock_t' (aka 'unsigned long') to 'unsigned int' [-Werror,-Wshorten-64-to-32]
                srand(clock());
                ~~~~~ ^~~~~~~
1 error generated.
````
**I've fixed this an other compilation errors** in the latest code from Development with PR #13982

3. When building with the "MacBook Air 2020" machine, which has the **M1 chipset, remember to add the option `-DCMAKE_OSX_ARCHITECTURES=x86_64`** to the `cmake .. -G "Xcode ...` command, which it's not specified in the build steps in the description.

4.  The "Editor.PhysXSupport5.c885bb7.building.but.not.launching.prior.to.update.terminal.log.MacBook.Air.txt" is crashing with the following assert:
`````
AssetBuilder: System: Trace::Assert
AssetBuilder:  /Users/macpro15/o3de/Code/Framework/AzCore/AzCore/Memory/HphaAllocator.cpp(921): (8800749056) 'void AZ::HphaSchemaBase<false>::HpAllocator::free(void *, size_t, size_t) [DebugAllocator = false]'
AssetBuilder: System: HPHA Assert, expression: "ptr_in_bucket(ptr)", possible reason: "small object ptr not in a bucket"
AssetBuilder: E: HPHA Assert, expression: "ptr_in_bucket(ptr)", possible reason: "small object ptr not in a bucket"
AssetBuilder: : AZ::HphaSchemaBase<false>::deallocate(void*, unsigned long, unsigned long) (+0xb5) [0x1357ed755]
AssetBuilder: : AZStd::vector<AZStd::pair<AZ::BehaviorMethod*, AZ::BehaviorClass*>, AZStd::allocator>::push_back(AZStd::pair<AZ::BehaviorMethod*, AZ::BehaviorClass*>&&) (+0x1a6) [0x19b7bc4a6]
AssetBuilder: : AZ::BehaviorClass::PostProcessMethod(AZ::BehaviorContext*, AZ::BehaviorMethod&) (+0x52a) [0x19b7bc22a]
AssetBuilder: : AZ::BehaviorContext::ClassBuilder<AZStd::vector<AZ::Render::MaterialAssignmentId, AZStd::allocator> >::~ClassBuilder() (+0x77) [0x19d2bbbf7]
AssetBuilder: : AZ::OnDemandReflection<AZStd::vector<AZ::Render::MaterialAssignmentId, AZStd::allocator> >::Reflect(AZ::ReflectContext*) (+0x3680) [0x19d299420]
AssetBuilder: : AZ::ReflectContext::ExecuteQueuedOnDemandReflections() (+0x129) [0x19b7a4cb9]
AssetBuilder: : AZ::BehaviorContext::ClassBuilder<AZStd::unordered_map<AZ::Render::MaterialAssignmentId, AZ::Render::MaterialAssignment, AZStd::hash<AZ::Render::MaterialAssignmentId>, AZStd::equal_to<AZ::Render::MaterialAssignmentId>, AZStd::allocator> >::~ClassBuilder()
AssetBuilder: : AZ::ReflectContext::ExecuteQueuedOnDemandReflections() (+0x129) [0x19b7a4cb9]
AssetBuilder: : AZ::BehaviorContext::GlobalPropertyBuilder::~GlobalPropertyBuilder() (+0x25) [0x19b7b9235]
AssetBuilder: : AZ::Render::MaterialAssignment::Reflect(AZ::ReflectContext*) (+0xd6d) [0x19d20eead]
AssetBuilder: : AZ::Render::CommonSystemComponent::Reflect(AZ::ReflectContext*) (+0xc3) [0x19d0a9043]
AssetBuilder: : AZ::ReflectionManager::Reflect(AZ::Uuid, AZStd::function<void (AZ::ReflectContext*)> const&) (+0x129) [0x104fd9e29]
AssetBuilder: : AZ::ComponentApplication::RegisterComponentDescriptor(AZ::ComponentDescriptor const*) (+0x77) [0x1050da8a7]
AssetBuilder: : void AZ::Internal::EBusContainer<AZ::ComponentApplicationRequests, AZ::ComponentApplicationRequestsEBusTraits, (AZ::EBusAddressPolicy)0, (AZ::EBusHandlerPolicy)0>::Dispatcher<AZ::EBus<AZ::ComponentApplicationRequests, AZ::ComponentApplicationRequestsEBusT
AssetBuilder: : AZ::Outcome<void, AZStd::basic_string<char, AZStd::char_traits<char>, AZStd::allocator> > AZStd::Internal::function_util::get_invoker<AZ::Outcome<void, AZStd::basic_string<char, AZStd::char_traits<char>, AZStd::allocator> > (), AZStd::Internal::function_u
AssetBuilder: : AZ::ModuleManager::LoadDynamicModules(AZStd::vector<AZ::DynamicModuleDescriptor, AZ::AZStdAlloc<AZ::OSAllocator> > const&, AZ::ModuleInitializationSteps, bool) (+0xb7) [0x104fb51e7]
AssetBuilder: : void AZ::Internal::EBusContainer<AZ::ModuleManagerRequests, AZ::ModuleManagerRequests, (AZ::EBusAddressPolicy)0, (AZ::EBusHandlerPolicy)0>::Dispatcher<AZ::EBus<AZ::ModuleManagerRequests, AZ::ModuleManagerRequests> >::BroadcastResult<AZStd::vector<AZ::Outc
AssetBuilder: : AssetBuilderApplication::StartCommon(AZ::Entity*) (+0x194) [0x104a0dc84]
AssetBuilder: : AzToolsFramework::ToolsApplication::Start(AZ::ComponentApplication::Descriptor const&, AZ::ComponentApplication::StartupParameters const&) (+0x1c) [0x1057d59bc]
AssetBuilder: : main (+0xf2) [0x104a5f6f2]
`````
It seems to be failing very soon allocating memory with the `HphaAllocator`. Maybe it's a combination of the previous points mentioned what lead to this, I'm not sure. But **let's focus on the build errors first for this ticket**, then after closing this ticket if the editor keeps crashing then let's open a different GHI for it.

----

In summary, I think PR #13982 is the fix for this ticket.

In the meantime if your purpose is to test PhysX 5 you can unblock yourself by not building everything, so instead of specifying `--target ALL_BUILD` you can use `--target Editor` and that is enough to build everything needed for running the Editor. I was able to build the Editor target on a Mac with Development (446de8ac), which it already includes PhysX 5 (remember to pass the `-DAZ_USE_PHYX5=ON` option to the CMake configure command to enable it)

Thanks,
Aaron

**Comment by LB-MateuszSzweda** [comment]
Unfortunately the building failure due to failure to compile in target 'AzFramework.Tests' from project 'O3DE' still occurs on my
MacBook Air 2020 M1 chipset machine on latest Development [7dee349] build when target **ALL_BUILD** is used.

Current MacBook Air 2020 M1 configuration is:  **M1 chipset, Ventura 13.0.1 , XCode 14.2, Cmake 3.25.0**

Regarding the comments to that particular machine.
1. It indeed had permission errors, they were likely to be introduced with one of OSX updates. I've failed to fix that with disk utility first aid and manually allowing _sudo_ privileges to user, so I have performed mac erase and OSX system setup from the scratch which resolved the permission error.

2. I've refrained from using _-DCMAKE_OSX_ARCHITECTURES=x86_64_ argument originally as comment to https://github.com/o3de/o3de/issues/13010 suggested it's no longer needed, but I've verified the above building error occurs both with and without _-DCMAKE_OSX_ARCHITECTURES=x86_64_  argument.

The commands I've used this time to configure and build were:

`cmake .. -G "Xcode" -DLY_3RDPARTY_PATH=$LY_3RDPARTY_PATH -DLY_PROJECTS=AutomatedTesting -DLY_UNITY_BUILD=1 -DCMAKE_OSX_ARCHITECTURES=x86_64 -DAZ_USE_PHYSX5=ON`

`cmake -E time cmake --build . --target ALL_BUILD --config profile`

The building error goes as follows:
```
** BUILD FAILED **
The following build commands failed:
	CompileC /Users/macpro/Desktop/build/o3de/osx_build/build/AzCore.Tests.build/profile/Objects-normal/x86_64/unity_1_cxx.o /Users/macpro/Desktop/build/o3de/osx_build/Code/Framework/AzCore/CMakeFiles/AzCore.Tests.dir/Unity/unity_1_cxx.cxx normal x86_64 c++ com.apple.compilers.llvm.clang.1_0.compiler (in target 'AzCore.Tests' from project 'O3DE')
(1 failure)
```

As a further note I'm including detailed steps of how the machine and build were setup below:

1. Machine was cleaned of every content and user with **System Settings > General > Erase All Content and Settings**
2. MacOS **Ventura 13.0.1**  was installed and new admin user was created.
3. After logging in with AppleID, **XCode 14.2** was installed via App Store.
4. Homebrew was obtained from https://brew.sh/ and `brew install git-lfs` was used to install git-lfs.
5. Development branch of o3de repository was cloned onto the machine.
6. _Git fetch_ , _Git merge_ and _git log_ commands were used to make sure that the obtained build indeed has the latest possible commit.
6. "3rdParty" and "osx_build" folders were created in the cloned repository folder.
7. Cmake 3.25.0 was moved into 3rdParty folder.
8. o3de python folder was opened in terminal, paths to Cmake and 3rdParty folder were provided and ./get_python.sh command was used to obtain packages.
9. As the ./get_python.sh failed with _Bad CPU type in executable_ which was pinpointed to M1 related issue https://apple.stackexchange.com/questions/408375/zsh-bad-cpu-type-in-executable
`softwareupdate --install-rosetta` was used and terminal app was briefly set to _Open using Rosetta_. After that the 
./get_python.sh correctly downloaded the packages and continued to do so even after _Open using Rosetta_ was disabled again.
10. After disabling _Open using Rosetta_ in terminal and having obtained packages with ./get_python "osx_build" folder was opened with terminal and following paths were set with:
export PATH=$PATH:/Users/<home_folder_name>/o3de/3rdParty/CMake/CMake.app/Contents/bin
export LY_3RDPARTY_PATH=/Users/<home_folder_name>/o3de/3rdParty
export LY_PACKAGE_SERVER_URLS=https://d2c171ws20a1rv.cloudfront.net
11. Then configuration was performed with:
cmake .. -G "Xcode" -DLY_3RDPARTY_PATH=$LY_3RDPARTY_PATH -DLY_PROJECTS=AutomatedTesting -DLY_UNITY_BUILD=1 -DCMAKE_OSX_ARCHITECTURES=x86_64 -DAZ_USE_PHYSX5=ON
12. After configuring building was performed with:
cmake -E time cmake --build . --target ALL_BUILD --config profile

For further information please refer to the configuration, building, git, get_python and failed get python (without rosseta installed) logs attached to this comment.

[1.12  Macbook Air M1 failed get python log (without rosetta installed).txt](https://github.com/o3de/o3de/files/10402604/1.12.Macbook.Air.M1.failed.get.python.log.without.rosetta.installed.txt)

[1.12  Macbook Air M1 configure log.txt](https://github.com/o3de/o3de/files/10402614/1.12.Macbook.Air.M1.configure.log.txt)

[1.12  Macbook Air M1 failed building log.txt](https://github.com/o3de/o3de/files/10402625/1.12.Macbook.Air.M1.failed.building.log.txt)

[1.12  Macbook Air M1 get python log.txt](https://github.com/o3de/o3de/files/10402638/1.12.Macbook.Air.M1.get.python.log.txt)

[1.12  Macbook Air M1 git log.txt](https://github.com/o3de/o3de/files/10402639/1.12.Macbook.Air.M1.git.log.txt)

Also please note that --target Editor built successfully using the same configuration.

**Comment by moraaar** [comment]
I see the build error in "1.12 Macbook Air M1 failed building log.txt". I'll put up a PR to fix it now.

For the error of "1.12 Macbook Air M1 failed get python log (without rosetta installed).txt", that's a separate issue, please open a different GHI for it.

**Comment by LB-MateuszSzweda** [comment]
Issue verified as fixed in Development (1cb47cc) . MacBook Air 2020 with M1 chipset correctly built ALL BUILD configuration.

https://github.com/o3de/o3de/issues/13010 was encountered but as stated in the issue, rerunning the `cmake -E time cmake --build . --target ALL_BUILD --config profile` command resulted in successful build.
